pkg/eval/vals/equal.go

```diff
@@ -22,7 +25,11 @@ func Equal(x, y interface{}) bool {
 	case bool:
 		return x == y
 	case float64:
-		return x == y
+		if math.Signbit(x) == math.Signbit(y.(float64)) &&
```

Comment:
Let's just stick to Go's standard `==` semantics here.

Reply:
We can't because your code does not stick to Go's standard `==` semantics. The code I added is compatible with Go's standard `==` semantics for infinities while still supporting the +/- epsilon of your code. Without this change if I replace your

```
return math.Abs(x-y.(float64)) <= float64EqualityThreshold
```

with

```
return x == y.(float64)
```

then my unit test passes but this failure occurs:

```
ok      github.com/elves/elvish/pkg/eval/daemon 0.300s
--- FAIL: TestMath (0.09s)
    --- FAIL: TestMath/math:sin_3.141592653589793 (0.00s)
        testutils.go:143: got value out [1.2246467991473515e-16], want [0]
    --- FAIL: TestMath/math:tan_1 (0.00s)
        testutils.go:143: got value out [1.557407724654902], want [1.5574077246549023]
    --- FAIL: TestMath/math:tan_3.141592653589793 (0.00s)
        testutils.go:143: got value out [-1.2246467991473515e-16], want [0]
```

Reply:
/facepalm :smile: I'm the one who added the epsilon test. Okay, I'll revert that portion of the change.